crates/sui-types/src/messages.rs

```diff
@@ -2357,6 +2376,22 @@ pub struct InvalidSharedByValue {
     pub object: ObjectID,
 }
 
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize, Hash)]
+pub enum CommandArgumentError {
```

Comment:
Would benefit from some docs:

- An overall doc comment reminding folks that these come from programmable transaction commands.
- A short description of how these errors come about (especially for the `Invalid*` errors -- what makes those things invalid?)

Reply:
(Or alternatively, a pointer in the outer doc comment to say that the `Display` implementation has this information 
 -- is there any chance we could use `thiserror` for this to bring these two things together?).

Reply:
Oooo fancy, will do